crates/router/tests/connectors/trustpay.rs

```diff
@@ -0,0 +1,304 @@
+    utils::{self, ConnectorActions},
+};
+
+#[derive(Clone, Copy)]
+struct TrustpayTest;
+impl ConnectorActions for TrustpayTest {}
+impl utils::Connector for TrustpayTest {
+    fn get_data(&self) -> types::api::ConnectorData {
+        use router::connector::Trustpay;
+        types::api::ConnectorData {
+            connector: Box::new(&Trustpay),
+            connector_name: types::Connector::Trustpay,
+            get_token: types::api::GetToken::Connector,
+        }
+    }
+
+    fn get_auth_token(&self) -> types::ConnectorAuthType {
+        types::ConnectorAuthType::from(
+            connector_auth::ConnectorAuthentication::new()
+                .trustpay
+                .expect("Missing connector authentication configuration"),
+        )
+    }
+
+    fn get_name(&self) -> String {
+        "trustpay".to_string()
+    }
+}
+
+fn get_default_browser_info() -> BrowserInformation {
+    BrowserInformation {
+        color_depth: 24,
+        java_enabled: false,
+        java_script_enabled: true,
+        language: "en-US".to_string(),
+        screen_height: 1080,
+        screen_width: 1920,
+        time_zone: 3600,
+        accept_header: "*".to_string(),
+        user_agent: "none".to_string(),
+        ip_address: None,
+    }
+}
+
+fn get_default_payment_authorize_data() -> Option<types::PaymentsAuthorizeData> {
+    Some(types::PaymentsAuthorizeData {
+        payment_method_data: types::api::PaymentMethodData::Card(api::Card {
+            card_number: Secret::new("4200000000000000".to_string()),
+            card_exp_year: Secret::new("25".to_string()),
+            card_cvc: Secret::new("123".to_string()),
+            ..utils::CCardType::default().0
+        }),
+        browser_info: Some(get_default_browser_info()),
+        ..utils::PaymentAuthorizeType::default().0
+    })
+}
+
+fn get_default_payment_info() -> Option<utils::PaymentInfo> {
+    Some(utils::PaymentInfo {
+        address: Some(types::PaymentAddress {
+            billing: Some(api::Address {
+                address: Some(api::AddressDetails {
+                    first_name: Some(Secret::new("first".to_string())),
+                    last_name: Some(Secret::new("last".to_string())),
+                    line1: Some(Secret::new("line1".to_string())),
+                    line2: Some(Secret::new("line2".to_string())),
+                    city: Some("city".to_string()),
+                    zip: Some(Secret::new("zip".to_string())),
+                    country: Some("IN".to_string()),
+                    ..Default::default()
+                }),
+                phone: None,
+            }),
+            ..Default::default()
+        }),
+        router_return_url: Some(String::from("http://localhost:8080")),
+        ..Default::default()
+    })
+}
+
+static CONNECTOR: TrustpayTest = TrustpayTest {};
+
+// Cards Positive Tests
+// Creates a payment using the automatic capture flow (Non 3DS).
+#[actix_web::test]
+async fn should_make_payment() {
```

Comment:
If the tests are working, please add a screenshot of the test result in the PR description if possible

Reply:
Also update the connector credentials in [this sheet](https://docs.google.com/spreadsheets/d/15VHQYJBkQnzqDXK66d7mtVuyQscNAHfCqnTFI-pCZY8/edit?pli=1#gid=160500608&range=A1)

Reply:
Sure, will do that